[CS2113T-W11-3] NUS Buddy

richwill28 opened the pull request:
NUS Buddy is a simple desktop app for managing your todo list. It is optimized for use via a Command Line Interface (CLI) so students can spend most of their time using the terminal and focus on actual works. It is designed to be fast, flexible, and unobtrusive.

alvintan01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+**Main components of the architecture**
+
+`Main` has two classes called `Duke` and `DukeException`.
+
+`Duke` is responsible for:
+* **At app launch:** Initializes the components in the correct sequence, and connects them up with each other.
+* **At shut down:** Shuts down the components and invokes cleanup methods where necessary.
+
+`DukeException` is responsible for:
+* Manages Duke-related exceptions.
+
+`Commons` represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
+* `UI`: The UI of the App.
+* `Logic`: The parser and command executor.
+* `Model`: Holds the various data structures of the App.
+* `Storage`: Reads data from, and writes data to, the hard disk.
```
Perhaps it's better to add bullet points for each of the main components?

richwill28 replied:
Hi @alvintan01 thanks for the review. Could you elaborate more on what you meant by *each components*?

alvintan01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+![StorageLoadSequence](images/LoadDataSequenceDiagram.png)
+
+1. A FileReader and created and subsequently used to create a BufferedReader that points to the given path.
```
Perhaps you meant "A FileReader is created..."?

richwill28 replied:
Thanks for pointing out the mistake.

alvintan01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+### Loading data
+
+![StorageLoadSequence](images/LoadDataSequenceDiagram.png)
```
Perhaps it is better to add a note to the diagram to show the possible exceptions thrown? For example, "loadData throws IOException when it is unable to read from the file"

alvintan01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+![LogicComponentDiagram](images/LogicComponentDiagram.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+> 📝 **Note:** XYZ is a placeholder for specific command name (e.g., AddModuleCommand).
```
Perhaps it's better to add this note to the diagram similar to the Logic component diagram in AB3?

richwill28 replied:
Good point.

alvintan01 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+#### Task Component
+
+**API** [`Task.java`](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/master/src/main/java/seedu/duke/model/task/Task.java)
```
Should all the *TaskCommands depend on TaskList only? Then you could remove one set of dependencies?

JWweiyin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
-## User Stories
+**API** : [`Lesson.java`](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/master/src/main/java/seedu/duke/model/lesson/Lesson.java)
+
+![LessonComponentObject](images/LessonComponentDiagram.png)
```
Should crossed lines be avoided to prevent confusion to the reader?

richwill28 replied:
Hi @JWweiyin thanks for the review. Could you elaborate more on what you meant by *crossed lines*?

JWweiyin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+**API** [`Module.java`](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/master/src/main/java/seedu/duke/model/module/Module.java)
+
+![ModuleComponentObject](images/ModuleComponentDiagram.png)
```
Could the diagram utilise a different colour scheme in order to distinguish it from the diagram above and the diagram below? Also could the diagram be referenced in the text below ("Above is the component diagram for the Module component etc")?

JWweiyin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+### Saving data
+
+![StorageSaveSequence](images/SaveDataSequenceDiagram.png)
```
In the return arrow for serializedTaskList, should it have a type?

huien77 replied:
![image](https://user-images.githubusercontent.com/69332531/139387381-c54f294e-37ed-496e-abf5-d5ea97c63c91.png)

I don't think there should be entity at the bottom of sequence diagram. If I am not wrong, you can remove these entities using `hide footbox` in puml

JWweiyin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+![LogicComponentDiagram](images/LogicComponentDiagram.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+> 📝 **Note:** XYZ is a placeholder for specific command name (e.g., AddModuleCommand).
```
The use of the note here is helpful in letting the user understand the placeholder. The use of emoji in the markup grabs the grabs attention so that this piece of information does not go missed by the reader.

JWweiyin reviewed (commented):
A good attempt at explaining the software, some of the diagrams may be a bit confusing with too many crossed lines.

Roycius committed:
Add total mc printing to list module

Roycius committed:
Remove "grades", change "all" to "verbose"

richwill28 committed:
Update help command output

ptejasv committed:
Fix syntax to match coding standard

richwill28 committed:
Fix DG typo

ptejasv committed:
Update UG grammar

richwill28 committed:
Change constant name to uppercase

Roycius committed:
Fix grade displaying bug

huien77 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+The **Sequence Diagram** below shows how the components interact with each other for the scenario where the 
+user issues the command `add task homework -d mon`.
+
+![ArchitectureDemoSequenceDiagram](images/ArchitectureDemoSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/69332531/139385317-38a3269e-af35-49d7-894e-2b4ebc70a74e.png)

I think there shouldn't be dotted line after end of lifeline (like after the cross)? Although, I think the problem is with plantUML.

Roycius committed:
Add JUnit for getTotalMcs

huien77 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
-## User Stories
+**API** : [`Lesson.java`](https://github.com/AY2122S1-CS2113T-W11-3/tp/blob/master/src/main/java/seedu/duke/model/lesson/Lesson.java)
+
+![LessonComponentObject](images/LessonComponentDiagram.png)
```
![image](https://user-images.githubusercontent.com/69332531/139385898-56c96144-b97a-41e2-8e83-19f8d607de85.png)

I think the lines are a little messy and hard to understand. Maybe can split up the diagram?

richwill28 committed:
Update UI print message (#104)

Fixes #92

richwill28 committed:
Update UG to match coding standard (#107)

Fixes #97

huien77 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,266 @@
+
+**Main components of the architecture**
+
+`Main` has two classes called `Duke` and `DukeException`.
```
![image](https://user-images.githubusercontent.com/69332531/139388331-49866a29-d75d-4da5-943f-73b69e48a723.png)

You mentioned there are two classes `Duke` and `DukeException`, but I cannot find `DukeException` in the architecture diagram. Maybe you missed it?

huien77 reviewed (commented):
Small changes to make. Overall good. :)

Roycius committed:
Merge branch 'master' into feature/listmodule_refinement

# Conflicts:
#	src/main/java/seedu/duke/ui/Ui.java

Roycius committed:
List module refinement (#105)

Fixes #101 
Fixes #102 
Fixes #103

richwill28 committed:
Update UG to conform to the latest program output

richwill28 committed:
Fix help example output

richwill28 committed:
Update help command output and UG (#106)

Fixes #93

Roycius committed:
Refactor AddCommandParser

Roycius committed:
Add JUnit

richwill28 committed:
Implement link utility class

richwill28 committed:
Add launch lesson feature

richwill28 committed:
Remove invalid link validation

richwill28 committed:
Add JUnit test for LinkUtil

richwill28 committed:
Add empty link validation

richwill28 committed:
Update empty link validation

Roycius committed:
Move some methods to ParserUtil

Roycius committed:
Fix issue #117

richwill28 committed:
Add JUnit for launchUrl

Roycius committed:
Remove shell from UG

richwill28 committed:
Update launchUrl test

Roycius committed:
Fix issue #115

richwill28 committed:
Fix url not launching on WSL

richwill28 committed:
Add launch lesson link feature (#114)

Fixes #111

Roycius committed:
Revert "Fix issue #115"

This reverts commit ad9214067eb8e683b3a3d02000b73251a05f642c.

Roycius committed:
Code review fixes (#110)

Fixes #99 
Fixes #117 
Fixes #118, #119, #120, #121, #122, #124, #125, #126, #127, #128, #136, #144

rebc39 committed:
Change product name in README

rebc39 committed:
Correct some minor errors in UG

rebc39 committed:
Update UG to remove shell markup & fix minor spelling & grammar errors

rebc39 committed:
Update Ui print message method to remove duplicate printing of grade

rebc39 committed:
Update toString method for Module class to add seperator between MCs and grade

rebc39 committed:
Update UG expected output for modules

rebc39 committed:
Add section on save data format

rebc39 committed:
Merge branch 'master' into feature/ug-dg-ui-review

# Conflicts:
#	docs/UserGuide.md

ptejasv committed:
Print better message for empty lists

ptejasv committed:
Change error message for loading corrupt module data

ptejasv committed:
Update tests to match new messages

rebc39 committed:
Change grades array to be constant

ptejasv committed:
Refractor print messages

ptejasv committed:
Fix message printing bugs (#159)

Fixes #155 
Fixes #142

ptejasv committed:
Update print message for empty module list

ptejasv committed:
Update LessonList to disallow duplicate lessons

ptejasv committed:
Add tests to check AddLessonCommand

ptejasv committed:
Update code to follow guidelines

richwill28 committed:
Update method signature to conform to checkstyle

richwill28 committed:
Fix buggy day of the week validation

richwill28 committed:
Add JUnit test for day of the week

richwill28 committed:
Add reposense authorship

richwill28 committed:
Make "-v" a valid list verbose option

richwill28 committed:
Add StingUtil class

richwill28 committed:
Update print module methods

richwill28 committed:
Delete unused method

richwill28 committed:
Update input commands

- list module does not support extra params
- find module has verbose option

richwill28 committed:
Update module getInfo method

richwill28 committed:
Update JUnit tests

Roycius committed:
Improve prompts error messages

ptejasv committed:
Refractor list module command

ptejasv committed:
Refractor tests to emphasise happy path

Roycius committed:
Rename class CommandFormat

richwill28 committed:
Fix buggy day of the week validation (#163)

Fixes #149.

richwill28 committed:
Update module-related commands (#164)

- Replace `moduleinfo` with find module
- `list module` no longer support verbose option
- Verbose option is given to `find module`

Fixes #162

ptejasv committed:
Manage merge conflicts

ptejasv committed:
Disallow duplicate lessons (#161)

Fixes #160

rebc39 committed:
Prevent usage of divider for user input

rebc39 committed:
Merge branch 'master' into feature/ug-dg-ui-review

Roycius committed:
Merge branch 'master' into feature/improve-error-message

# Conflicts:
#	src/main/java/seedu/duke/logic/parser/AddCommandParser.java
#	src/main/java/seedu/duke/logic/parser/FindCommandParser.java
#	src/main/java/seedu/duke/logic/parser/ListCommandParser.java
#	src/main/java/seedu/duke/ui/Ui.java

Roycius committed:
Improve prompts for error messages (#165)

Fixes #141

rebc39 committed:
Merge branch 'master' into feature/ug-dg-ui

rebc39 committed:
Add CS and CU grades

rebc39 committed:
Move symbol checking and replacement to AddParser

rebc39 committed:
Add jar file name in UG

rebc39 committed:
Add product introduction from UG to README

rebc39 committed:
Remove footboxes for sequence diagrams

rebc39 committed:
Minor changes to UI, parser, UG, DG #166

Fixes #109 
Fixes #116 
Fixes #134

Roycius committed:
Change flags implementation to be flexible

Roycius committed:
Remove JUnit for old method

Roycius committed:
More OOP

Roycius committed:
Fix typos and simplify function slightly

Roycius committed:
Add JUnits tests

Roycius committed:
Extract flags to be constants in parser

Roycius committed:
Add JavaDoc and clean ParserUtil

Roycius committed:
Flag feature enhancement (#167)

Fixes #115, Fixes #123, Fixes #133, Fixes #151, Fixes #158

Roycius committed:
Rearrange ModuleComponentDiagram

Roycius committed:
Add DG for save file formatting

richwill28 committed:
Add PPP draft for richwill28

richwill28 committed:
Add profile picture

richwill28 committed:
Add profile picture in AboutUs.md

richwill28 committed:
Adjust picture size

Roycius committed:
Update Commons(core) class

Roycius committed:
Draft PPP skeleton

Roycius committed:
Minor typo fix

richwill28 committed:
Add PPP draft for richwill28 (#168)

Roycius committed:
Update DG (#169)

Fixes #96

ptejasv committed:
Add PPP skeleton for ptejasv

rebc39 committed:
Add initial draft of PPP

rebc39 committed:
Resolve merge conflict

ptejasv committed:
Update AboutUs portfolio links

ptejasv committed:
Add PPP skeleton for ptejasv (#170)

Fixes #171

rebc39 committed:
Merge branch 'master' into documentation/ppp-draft

rebc39 committed:
Add headings for PPP

rebc39 committed:
Move reposense link to subsection

rebc39 committed:
Add product overview

richwill28 committed:
Add PPP draft for rebchua39 (#172)

richwill28 committed:
Refactor Messages to Message for class naming consistency

richwill28 committed:
Rename variable to improve consisteny of code

richwill28 committed:
Standardize time to 24-hour format

richwill28 committed:
Remove vscode files

richwill28 committed:
Override new contructor for clarity

richwill28 committed:
Sort lesson by timeline

Roycius committed:
PPP update

Roycius committed:
Update new sections in DG

Roycius committed:
Update UG to cater to new commands

Roycius committed:
Add missing info in UG

Roycius committed:
Update command summary

Roycius committed:
Update verbose commands in UG

Roycius committed:
Fix #140 and example indentation

richwill28 committed:
Add JUnit for default task sorting

richwill28 committed:
Add JUnit for default lesson sorting

richwill28 committed:
Add JUnit for lesson compareTo

richwill28 committed:
Add exception handling for compareTime

richwill28 committed:
Add JUnit for DayUtil

richwill28 committed:
Add Junit for TimeUtil

Roycius committed:
Update PPP, UG and DG (#176)

Fixes #140 
Fixes #157

richwill28 committed:
Implement default lesson sorting (#173)

Fixes #156

Roycius committed:
Make extra trailing words invalid

Roycius committed:
Rename PPP to `roycius.md`

Roycius committed:
Add Validation for trailing words (#177)

Fixes #132, Fixes #174

rebc39 committed:
Fix bug in find task command parsing

rebc39 committed:
Prevent adding lessons with start time after end time

rebc39 committed:
Delete use cases section

rebc39 committed:
Add non-functional requirements

rebc39 committed:
Add terms in glossary section

rebc39 committed:
Add manual testing instructions for tasks

rebc39 committed:
Add manual testing instructions for lessons

rebc39 committed:
Add manual testing instructions for modules

rebc39 committed:
Add section links for testing instructions

rebc39 committed:
Add content for PPP

rebc39 committed:
Add manual testing section links in table of contents

rebc39 committed:
Remove use cases in table of contents

rebc39 committed:
Refactor error message for adding lesson timing

rebc39 committed:
Update DG and minor fixes (#178)

Fixes #113

rebc39 committed:
Merge branch 'master' into documentation/ppp-draft

rebc39 committed:
Update PPP details

richwill28 committed:
Add CommandFlag class

richwill28 committed:
Reorder variables according to alphabetical order

richwill28 committed:
Update addModule signature to include grade

richwill28 committed:
Fix array indentation error for checkstyle

richwill28 committed:
Add Grade class

richwill28 committed:
Refactor parser methods to StringUtil

richwill28 committed:
Fix parse grade bug

richwill28 committed:
Improve user command to be case-insensitive

richwill28 committed:
Fix wrong error message for blank input

richwill28 committed:
Refactor SetGradeCommand to EditModuleCommand

richwill28 committed:
Remove edit lessona and task parser

richwill28 committed:
Fix ModuleListTest

richwill28 committed:
Fix grade comment typo

richwill28 committed:
Fix broken link for roycius.md

richwill28 committed:
Remove unused format

richwill28 committed:
Remove unsupported format from parser

richwill28 committed:
Refactor SetGradeCommand to EditModuleCommand (#184)

Fixes #180 
Fixes #183 
Fixes #181 
Fixes #145 
Fixes #182

Roycius committed:
Fix helpcommand prompt for changing grade

Roycius committed:
Change setgrade to editmodule in UG

Roycius committed:
Merge branch 'fix/helpcommand-prompt' into documentation/update-documentation

Roycius committed:
Update "help" in UG

ptejasv committed:
Update ptejasv PPP

richwill28 committed:
Update overview and contributions

Roycius committed:
Update DG for Common classes

ptejasv committed:
Add profile image

ptejasv committed:
Add command descriptions

ptejasv committed:
Add tests to ParserTest

ptejasv committed:
Update PPP for ptejasv

Roycius committed:
Update PPP

Roycius committed:
Merge branch 'master' into documentation/update-documentation

# Conflicts:
#	docs/UserGuide.md

richwill28 committed:
Update team-based tasks

rebc39 committed:
Add profile picture

rebc39 committed:
Merge branch 'master' into documentation/ppp-draft

Roycius committed:
Add some JUnit tests

Roycius committed:
Fix typos in PP titles

richwill28 committed:
Update review/mentoring and PED bugs

Roycius committed:
Update documentation and "help" (#187)

rebc39 committed:
Remove extra symbol

rebc39 committed:
Resolve merge conflict

richwill28 committed:
Update code contributions

richwill28 committed:
Add more PRs for features improvement

richwill28 committed:
Resolve merge conflict

richwill28 committed:
Update overview

rebc39 committed:
Remove height attribute for picture

rebc39 committed:
Add content to PPP

richwill28 committed:
Update PPP for richwill28 (#188)

richwill28 committed:
Shorten PPP to conform to page limit

rebc39 committed:
Update PPP draft #190

richwill28 committed:
Add JavaDoc for CommandFlag

richwill28 committed:
Remove extra line

richwill28 committed:
Reorder methods + add JavaDoc

richwill28 committed:
Reorder methods

richwill28 committed:
Add Javadoc for DayOfTheWeek

richwill28 committed:
Add Javadoc for Grade

richwill28 committed:
Reorder imports

richwill28 committed:
Reorder imports

richwill28 committed:
Reorder imports

richwill28 committed:
Reorder imports

richwill28 committed:
Reorder imports

richwill28 committed:
Reorder imports

richwill28 committed:
Remove extra line

richwill28 committed:
Update comment

richwill28 committed:
Reorder imports

richwill28 committed:
Update exceptions and error messages for ModuleList

richwill28 committed:
Remove vscode files

richwill28 committed:
Add Javadoc for DayUtil

richwill28 committed:
Add JavaDoc for util classes

richwill28 committed:
Update error messages for model comparison + reorder imports

richwill28 committed:
Delete unused exception

richwill28 committed:
Add warning if fullmodulelist fail to load

richwill28 committed:
Reorder imports

richwill28 committed:
Update error message for module parsing

richwill28 committed:
Update height for richwill28 image

richwill28 committed:
Add logging section

richwill28 committed:
Update uml colorscheme

ptejasv committed:
Update DG diagrams

ptejasv committed:
Update DG diagrams

richwill28 committed:
Fix UML bugs

- Fix inconsistent color scheme
- Fix incorrect class diagram syntax

richwill28 committed:
Update diagram images

richwill28 committed:
Update description

richwill28 committed:
Fix coding standards, add more Javadoc, and refine messages (#192)

richwill28 committed:
Fix module print bugs for long description

richwill28 committed:
Update finf module output

richwill28 committed:
Fix grammar

richwill28 committed:
Update DG and standardize UML color scheme (#193)

richwill28 committed:
Merge branch 'master' into documentation/fix-PPP-length

richwill28 committed:
Add extracts for UG and DG

richwill28 committed:
Update PR list

richwill28 committed:
Shorten PPP for richwill28 (#191)

ptejasv committed:
Update Add command syntax

ptejasv committed:
Update UG find module command summary

ptejasv committed:
Remove extra spaces

ptejasv committed:
Update Add command syntax

richwill28 committed:
Fix period bugs

richwill28 committed:
Update print format

richwill28 committed:
Fix checkstyle

richwill28 committed:
Fix period bugs (#197)

ptejasv committed:
Update UG examples

ptejasv committed:
Update UG examples
